pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -236,23 +239,100 @@ func (r *ReconcileComplianceRemediation) reconcileRemediation(instance *compv1al
 				"Make sure the CRD is installed: %w", err)
 	} else if kerrors.IsNotFound(err) {
 		if instance.Spec.Apply {
+			// Going through remediation list, to make sure all the related
+			// remediations objects are set to apply
+			err := r.setRemediationsToApply(instance, objectLogger)
+			if err != nil {
+				objectLogger.Error(err, "Failed to set related remediations to apply")
+				return err
+			}
+			err = r.createRemediation(obj, objectLogger)
+			if err != nil {
+				return err
+			}
 			instance.AddOwnershipLabels(obj)
-			return r.createRemediation(obj, objectLogger)
+			return nil
 		}
-
 		objectLogger.Info("The object wasn't found, so no action is needed to unapply it")
 		return nil
 	} else if err != nil {
 		return err
 	}
 
 	if instance.Spec.Apply {
+		err = r.setRemediationsToApply(instance, objectLogger)
+		if err != nil {
+			objectLogger.Error(err, "Failed to set related remediations to apply")
+			return err
+		}
 		return r.patchRemediation(obj, objectLogger)
 	}
-
+	err = r.setRemediationsToUnapply(instance, objectLogger)
+	if err != nil {
+		objectLogger.Error(err, "Failed to set related remediations to unapply")
+		return err
+	}
 	return r.deleteRemediation(obj, found, objectLogger)
 }
 
+// find all the other related remediation and set the apply to true on them
+func (r *ReconcileComplianceRemediation) setRemediationsToApply(instance *compv1alpha1.ComplianceRemediation, logger logr.Logger) error {
+	remediations := &compv1alpha1.ComplianceRemediationList{}
+	err := r.Client.List(context.TODO(), remediations, client.InNamespace(instance.GetNamespace()))
+	if err != nil {
+		return fmt.Errorf("couldn't list remediations: %w", err)
+	}
+	for _, remediation := range remediations.Items {
```

Comment:
maybe we could reduce code duplication between the two functions somehow?
either return a filtered list with only the matching names, or split a helper function that would accept a closure as an argument and the closure would be in this case:
```
if !remediation.Spec.Apply {
				remediation.Spec.Apply = true
				err = r.Client.Update(context.TODO(), &remediation)
				if err != nil {
					return fmt.Errorf("couldn't update remediation: %w", err)
				}
			}
```
and in the unapply case the opposite

Reply:
thanks for the reviews! that's very good idea, let me fix it